core/lib/dal/src/connection/mod.rs

```diff
@@ -24,12 +27,20 @@ impl fmt::Debug for ConnectionPoolBuilder {
         formatter
             .debug_struct("ConnectionPoolBuilder")
             .field("max_size", &self.max_size)
+            .field("acquire_timeout", &self.acquire_timeout)
             .field("statement_timeout", &self.statement_timeout)
             .finish()
     }
 }
 
 impl ConnectionPoolBuilder {
+    /// Sets the acquire timeout for a single connection attempt. There are multiple attempts (currently 3)
+    /// before `access_storage*` methods return an error.
+    pub fn set_acquire_timeout(&mut self, timeout: Duration) -> &mut Self {
```

Comment:
We could theoretically allow configuring this timeout via node config. For now, it's only called by test pools.

Reply:
I understand it's set to 30 seconds by defau;lt? I think this is a very large number. I feel like it should be something like 1 second, so def a very good idea to have this in config.
Although I'm not sure how most of the components react when they receive error when getting connection. Probably panic? But either way, I feel like we want to give up way earlier. If for API server it takes that much time to get a DB connection, we can assume that the system health is poor enough that we can just respond user with `500`. A more fundamental solution would be to introduce DB health to the circuit breaker.

Reply:
I'll introduce the corresponding config parameter in a follow-up PR then.